### PR TITLE
Fix validation schemas

### LINKED ISSUE #423: Fix inconsistent validation
The default, min, and max values for the endpoints are handled inconsistently currently. Update the schemas.
Default values must be handled by the schema validation and not in lower layers (model, repository etc.) .
Update the docs and tests to reflect the changes.

### PR DESCRIPTION
**Pull request by pranavkparti**
## Description

The endpoints for `master` branch now have proper validation for query parameters like `limit`, `offset`, etc.  

**Issue(s) addressed**

- Resolves #423 

**What kind of change(s) does this PR introduce?**

- [ ] Enhancement
- [X] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**

The min, max, default values are not defined properly for many endpoints. The default values are also handled inconsistently.

**What is the new behavior?**

The validation schemas are updated.

## Breaking change

**Does this PR introduce a breaking change?**
No.

## Other useful information
No.

### CODE REVIEW

File: server/handlers/authHandler/schemas.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const Joi = require('joi');
 
 const authPostSchema = Joi.object({
-  wallet: Joi.string().min(4).max(36).required(),
+  wallet: Joi.alternatives().try(Joi.string(), Joi.string().uuid()).required(),
```
Comment (Kpoke) [inline server/handlers/authHandler/schemas.js:4]: we are not going to accept uuid as an option here

File: server/handlers/tokenHandler/schemas.js [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
-  offset: Joi.number().min(0).integer().default(0),
+  limit: Joi.number().integer().min(1).max(2000).default(2000),
+  offset: Joi.number().integer().min(0).default(0),
   wallet: Joi.string(),
 });
 
 const tokenGetTransactionsByIdSchema = Joi.object({
-  limit: Joi.number().min(1).max(1000).integer().default(1000).required(),
-  offset: Joi.number().min(0).integer(),
-  id: Joi.string().guid(),
```
Comment (Kpoke) [inline server/handlers/tokenHandler/schemas.js:12]: removed the id?

**Review by Kpoke — CHANGES_REQUESTED** [review]

File: server/handlers/tokenHandler/schemas.js (hunk 2 above)
Comment (pranavkparti) [reply server/handlers/tokenHandler/schemas.js:12]: Yes. The ID of the token is read from `req.params`. The `id` here in `req.query` is not needed. It is never used.

File: server/handlers/authHandler/schemas.js (hunk 1 above)
Comment (pranavkparti) [reply server/handlers/authHandler/schemas.js:4]: Currently AuthService uses wallet service's `getByIdOrName` to fetch the wallet. Should that be changed?

Comment (Kpoke) [reply server/handlers/authHandler/schemas.js:4]: yeah, we can change it to use just a getByName function

**Pull request merged**